### PR TITLE
refactor(emqx_cm): Decorate RPCs

### PR DESCRIPTION
**Pull request by k32**

### CODE REVIEW

**Review by qzhuyan — APPROVED** [review]

**Review by thalesmg — APPROVED** [review]

File: apps/emqx/src/emqx_cm.erl [hunk 1]
```diff
@@ -80,9 +80,15 @@
         , mark_channel_connected/1
         , mark_channel_disconnected/1
         , get_connected_client_count/0
+
```
Comment (mononym) [inline apps/emqx/src/emqx_cm.erl:83]: Extra/empty line for a reason?

Comment (k32) [reply apps/emqx/src/emqx_cm.erl:83]: Yes, to logically separate a group of functions.

File: apps/emqx/src/proto/emqx_cm_proto_v1.erl [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-module(emqx_cm_proto_v1).
+
+-behaviour(emqx_bpapi).
+
+-export([ introduced_in/0
+
```
[the remark this answers is not in the document]
Comment (k32) [reply apps/emqx/src/proto/emqx_cm_proto_v1.erl:22]: Yes, `introduced_in` is a behavior callbacks, while the rest of the functions are part of the API.

File: apps/emqx/test/emqx_cm_SUITE.erl [hunk 3]
```diff
@@ -283,19 +283,17 @@ flush_emqx_pool() ->
...
+       #{timetrap => 60000},
        begin
          #{conninfo := ConnInfo0} = ?ChanInfo,
          ConnInfo = ConnInfo0#{conn_mod := emqx_ws_connection},
          {Pid, Ref} = spawn_monitor(fun() -> receive stop -> exit(normal) end end),
          ok = emqx_cm:register_channel(ClientId, Pid, ConnInfo),
          Pid ! stop,
          receive {'DOWN', Ref, process, Pid, normal} -> ok end,
-         ok = emqx_cm:discard_session(ClientId),
-         {ok, _} = ?block_until(#{?snk_kind := "session_already_gone", pid := Pid}, 1000)
```
Comment (k32) [inline apps/emqx/test/emqx_cm_SUITE.erl:294]: There is a case when this event is not emitted.

**Comment by k32** [comment]
Elixir check failed due to network issue during dependency downloading.

**Pull request merged**